Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -227,8 +227,10 @@ namespace AzToolsFramework
                         }
                         sharedColumnLayout->addItem(itemAt(layoutIndex + sharedWidgetIndex));
 
-                        // If a widget should be expanded, stretch it
-                        if (itemAt(layoutIndex + sharedWidgetIndex)->minimumSize().width() > 18)
+                        //! Stretch labels and large widgets to fill the space they are given,
+                        //! overrides alignment since the stretched widget acts as a spacer
+                        auto* elidingLabel = qobject_cast<AzQtComponents::ElidingLabel*>(currentWidget);
+                        if (itemAt(layoutIndex + sharedWidgetIndex)->minimumSize().width() > 18 || elidingLabel)
```

Comment:
where is this 18 coming from? might want to define it as a constant variable.

Reply:
ooh yeah, please de-magic this number with a constexpr constant and a comment as to why we're using that value

Reply:
Changed it to constexpr. 18 is the width of ContainerActionButtons, small widgets like these that don't fill their column shouldn't get priority to expand in a shared column. If a user wants to give priority to a small widget, they can achieve that behavior by having the widget outside of a shared column and use the alignment attribute.

I'm not sure if this is the best way to check for small widgets, but I didn't think it was scalable to create a list of large widgets we should check to expand, or small widget list we should check to not expand. I also thought to try by checking if the horizontal size policy was expanding, but that did not work either since all widgets have that. I'm open to any suggestions :)